src/FileProcessor/Yaml/Form/FormYamlFileProcessor.php

```diff
@@ -54,15 +65,27 @@ public function process(File $file, Configuration $configuration): void
 
         // Nothing has changed. Early return here.
         if ($newYaml === $yaml) {
-            return;
+            return $systemErrorsAndFileDiffs;
         }
 
-        $newFileContent = Yaml::dump($newYaml, 99);
+        $spaceCount = $this->yamlIndentResolver->resolveIndentSpaceCount($oldYamlContent);
+
+        $newFileContent = Yaml::dump($newYaml, 99, $spaceCount);
         $file->changeFileContent($newFileContent);
```

Comment:
Before, the YAML was printed with default 2 space indent. That broke the tabs spacing and returned false positive.

The `YamlIndentResolver` service now makes sure the original indent is always kept. The regular expression might need some tweaks, but the main idea is finished :slightly_smiling_face:

Reply:
Thanks a lot men. Looks promising.